### PR TITLE
Add compatibility with RHEL7 (and YUM v3)

### PR DESCRIPTION
**Pull request by janblazek**
Related: https://issues.redhat.com/browse/OSCI-5194

### CODE REVIEW

**Review by jpopelka — APPROVED** [review]
Just the urls in the repo file, looks OK otherwise.
Sorry for late reply, I somehow didn't get notification about this PR, no idea why.

File: profiles/rhel/repos/rhel-7.9-z.repo [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+[rhel-RT-debuginfo]
+name=rhel-RT-debuginfo
+baseurl=http://download.eng.rdu2.redhat.com/released/RHEL-7/7.9/Server-RT/x86_64/debug/tree/
+enabled=1
+gpgcheck=0
+
+[rhel-updates]
+name=rhel-updates
+baseurl=http://download.eng.rdu2.redhat.com/rhel-7/rel-eng/updates/RHEL-7/RHEL-7.9-updates-20240305.0/compose/Server/x86_64/os/
```
[the remark this answers is not in the document]
Comment (janblazek) [reply profiles/rhel/repos/rhel-7.9-z.repo]: Thank you for noticing the wrong compose links!

**Comment by jpopelka** [comment]
@janblazek you can merge this anytime. Or do you want me to do it?

**Comment by AdamWill** [comment]
I think this may have broken all installability tests for Fedora PRs, e.g. http://artifacts.dev.testing-farm.io/ef9fb00a-212e-4bc9-a51b-5fc1c9b945a6 from https://src.fedoraproject.org/rpms/anaconda/pull-request/151# .

It changes the command used to find packages from `"$YUMDNFCMD" repoquery` to just plain `repoquery`. It looks like the `repoquery` command in current Fedoras at least is a thin wrapper around `dnf repoquery` that's provided by the `dnf-utils` package, which nothing in particular depends on, and I'm guessing it's not installed into the environment that's used for the installability tests on PRs (I'm not sure where that's defined).

This doesn't seem to be affecting the installability test run on Koji package builds, at least I see many passed results for that version of the test. I guess dnf-utils happens to be present in that environment, or something.

**Comment by AdamWill** [comment]
Additionally, it seems to have missed changing one `repoquery` call - there's still a `"$YUMDNFCMD" repoquery` in `mtps-get-task`.

I checked and confirmed that `yum repoquery` does not exist as a subcommand in EL 7, so I see why the change was made.

Weirdly, a run on a similar PR three hours later worked: https://artifacts.dev.testing-farm.io/a221e399-1a3a-419f-bacc-39e6a53988e7/ from https://src.fedoraproject.org/rpms/anaconda/pull-request/152 . I can't see what the difference between the two runs was. Did anyone do anything intentional to fix this? You can tell the `repoquery` command worked on the second but failed on the first by comparing the "# INSTALL TEST" output. From the failed test:

    12:16:24                 out:  ----------------
    12:16:24                 out: |                |
    12:16:24                 out: | # INSTALL TEST |
    12:16:24                 out: |                |
    12:16:24                 out: | Packages:      |
    12:16:24                 out: |                |
    12:16:24                 out: |                |
    12:16:24                 out: ----------------
    12:16:24                 out: Test 'install' failed due no packages for testing in repo 'brew-116502585:'.

From the passed test:

    15:08:01                 out:  -------------------------------------------------------------
    15:08:01                 out: |                                                             |
    15:08:01                 out: | # INSTALL TEST                                              |
    15:08:01                 out: |                                                             |
    15:08:01                 out: | Packages:                                                   |
    15:08:01                 out: |                                                             |
    15:08:01                 out: |    * anaconda-0:41.12-1.fc41.x86_64                         |
    15:08:01                 out: |    * anaconda-core-0:41.12-1.fc41.x86_64                    |
    15:08:01                 out: |    * anaconda-core-debuginfo-0:41.12-1.fc41.x86_64          |
    15:08:01                 out: |    * anaconda-debuginfo-0:41.12-1.fc41.x86_64               |
    15:08:01                 out: |    * anaconda-debugsource-0:41.12-1.fc41.x86_64             |
    15:08:01                 out: |    * anaconda-dracut-0:41.12-1.fc41.x86_64                  |
    15:08:01                 out: |    * anaconda-dracut-debuginfo-0:41.12-1.fc41.x86_64        |
    15:08:01                 out: |    * anaconda-gui-0:41.12-1.fc41.x86_64                     |
    15:08:01                 out: |    * anaconda-install-env-deps-0:41.12-1.fc41.x86_64        |
    15:08:01                 out: |    * anaconda-install-img-deps-0:41.12-1.fc41.x86_64        |
    15:08:01                 out: |    * anaconda-live-0:41.12-1.fc41.noarch                    |
    15:08:01                 out: |    * anaconda-tui-0:41.12-1.fc41.x86_64                     |
    15:08:01                 out: |    * anaconda-widgets-0:41.12-1.fc41.x86_64                 |
    15:08:01                 out: |    * anaconda-widgets-debuginfo-0:41.12-1.fc41.x86_64       |
    15:08:01                 out: |    * anaconda-widgets-devel-0:41.12-1.fc41.x86_64           |
    15:08:01                 out: |    * anaconda-widgets-devel-debuginfo-0:41.12-1.fc41.x86_64 |
    15:08:01                 out: |                                                             |
    15:08:01                 out: -------------------------------------------------------------

that box is populated with the content of `nevras_in_repo`, and `nevras_in_repo` is set by running `repoquery --repoid=${REPONAME} --all blahblahblah`. So clearly that failed in a test run at 12:16:24 but passed in the same test run on an almost-identical PR at 15:08:01. I don't know *why*, though.

I came up with two alternative options to fix this, but seems like it might be worth figuring out the mystery first...

**Comment by janblazek** [comment]
Thank you @AdamWill for catching this and @jpopelka for quick turn-around with new PR!
I'll try to be double cautious about testing on both RHEL and Fedora-CI pipeline next time.
Are you able to use previous copr build in the Fedora-CI pipeline in the meantime when something gets broken?

**Comment by AdamWill** [comment]
well, this time @jpopelka just deleted the broken build from COPR, which caused subsequent tests to use an older build.

**Pull request merged**